earth_enterprise/src/common/khMTTypes.h

```diff
@@ -258,88 +260,85 @@ class khMTMap
 // WARNING: Use this with extreme caution
 // It protects some obvious problems but creates some less obvious ones
 template <typename T>
-class MTVector {
+class MTVector : public std::vector<T>{
```

Comment:
Inheriting from std::vector makes me crazy nervous. We've already discussed that, though.

Reply:
Just for posterity, I'll register my agreement with @tst-lsavoie here and reiterate that the risks involved with this were a topic of a fair bit of discussion. However, since this code is considered experimental, with the understanding that anything that would be merged into master would need a more robust implementation, I'm okay with having it implemented this way in this branch.

Reply:
Also for posterity, here are some of the rationales that I considered before using this approach: 
1) Stroupstrup himself actually  subclassess std::vectors in    [https://isocpp.org/files/papers/4-Tour-Algo-draft.pdf](A Tour of C++: Containers and Algorithms) 
2) A prominent (but not prevalent) argument that it's OK so long as no state is maintained in the subclass.  This is the only way to address to the lack no virtual destructors in the std container classes - so we're good here.

Reply:
(also, makes me crazy nervous too!)